doc/api/tls.md

```diff
@@ -1696,12 +1696,16 @@ changes:
   * `sessionTimeout` {number} The number of seconds after which a TLS session
     created by the server will no longer be resumable. See
     [Session Resumption][] for more information. **Default:** `300`.
-  * `SNICallback(servername, cb)` {Function} A function that will be called if
-    the client supports SNI TLS extension. Two arguments will be passed when
-    called: `servername` and `cb`. `SNICallback` should invoke `cb(null, ctx)`,
-    where `ctx` is a `SecureContext` instance. (`tls.createSecureContext(...)`
-    can be used to get a proper `SecureContext`.) If `SNICallback` wasn't
-    provided the default callback with high-level API will be used (see below).
+  * `SNICallback(servername, callback)` {Function} A function that will be
+    called if the client supports SNI TLS extension. Two arguments will be
+    passed when called: `servername` and `callback`. `callback` is an
+    error-first callback that must be called synchronously and takes two
+    optional arguments: `error` and `ctx`. `ctx`, if provided, is a
+    `SecureContext` instance. [`tls.createSecureContext()`][] can be used to get
+    a proper `SecureContext`. If `callback` is called with a falsy `ctx`
+    argument, the default secure context of the server will be used. If
```

Comment:
Just to make sure -dId you chceck that this is what actually happens? I assumed it and shared my thought in the issue description, but never checked it inside code or with tests.
Also maybe it would make sense to add the test I've posted in the issue to the testsuite along with this commit.

Reply:
I did not test it but it seem to be true by reading https://github.com/nodejs/node/blob/f89530fccc9a2420cef58861aaafa9732683e154/lib/_tls_wrap.js#L211-L212

Reply:
Falsy is different from none. Please see my change request.

Reply:
> Next time please mind the "I would like to work on this issue and submit a pull request."

Oh sorry, I missed that. I'm fine with closing if you want to resubmit under your name. I don't care.

Reply:
That line in `_tls_wrap.js` was precisely what I quickly checked to make sure this was correct.

Reply:
I think a test is not needed because, if I'm not wrong, `test/parallel/test-tls-sni-option.js` already tests this behavior.

Reply:
Thanks, no need to, the comment was solely about avoiding redundant work.

Reply:
No error is emitted by this https://github.com/nodejs/node/blob/f89530fccc9a2420cef58861aaafa9732683e154/test/parallel/test-tls-sni-option.js#L111-L122